### PR TITLE
Added new keyboard boardsource/3x4

### PR DESCRIPTION
**Pull request by daysgobye**
## Description
added new folder for boardsource keyboards and added new keyboard (3x4) a small macro pad

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/boardsource/3x4/3x4.c [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// 	// runs for every action, just before processing by the firmware
+
+// 	return process_record_user(keycode, record);
+// }
+
+// void led_set_kb(uint8_t usb_led) {
+// 	// put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+
+// 	led_set_user(usb_led);
+// }
```
Comment (tzarc) [inline keyboards/boardsource/3x4/3x4.c:29]:
```suggestion
#include "3x4.h"
```

File: keyboards/boardsource/3x4/3x4.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+
+#include "quantum.h"
+// #define ___ KC_NO
+
+// This a shortcut to help you visually see your layout.
+// The first section contains all of the arguments
+// The second converts the arguments into a two-dimensional array
+
```
Comment (tzarc) [inline keyboards/boardsource/3x4/3x4.h:10]:
```suggestion
#include "quantum.h"
```

File: keyboards/boardsource/3x4/config.h [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+/*
+ * Keyboard Matrix Assignments
+ *
+ * Change this to how you wired your keyboard
+ * COLS: AVR pins used for columns, left to right
+ * ROWS: AVR pins used for rows, top to bottom
+ * DIODE_DIRECTION: COL2ROW = COL = Anode (+), ROW = Cathode (-, marked on diode)
+ *                  ROW2COL = ROW = Anode (+), COL = Cathode (-, marked on diode)
+ *
+*/
```
Comment (tzarc) [inline keyboards/boardsource/3x4/config.h:28]:
```suggestion
```

File: keyboards/boardsource/3x4/config.h [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+#define MATRIX_ROW_PINS { F7, F6, F5,}
+#define MATRIX_COL_PINS {B6, B2, B3, B1}
+
+/* COL2ROW, ROW2COL*/
```
Comment (tzarc) [inline keyboards/boardsource/3x4/config.h:32]:
```suggestion
```

File: keyboards/boardsource/3x4/config.h [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+// #define BACKLIGHT_PIN C7
+// #define BACKLIGHT_BREATHING
+// #define BACKLIGHT_LEVELS 3
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
```
Comment (tzarc) [inline keyboards/boardsource/3x4/config.h:39]:
```suggestion
```

File: keyboards/boardsource/3x4/config.h [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+#define DEBOUNCE 5
+
+/* define if matrix has ghost (lacks anti-ghosting diodes) */
+//#define MATRIX_HAS_GHOST
+
+/* number of backlight levels */
+
```
Comment (tzarc) [inline keyboards/boardsource/3x4/config.h:46]:
```suggestion
```

File: keyboards/boardsource/3x4/config.h [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 1
+
```
Comment (tzarc) [inline keyboards/boardsource/3x4/config.h:166]:
```suggestion
```

File: keyboards/boardsource/3x4/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+// }
+
+// bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+//   return true;
+// }
+
+// void led_set_user(uint8_t usb_led) {
+
+// }
```
Comment (tzarc) [inline keyboards/boardsource/3x4/keymaps/default/keymap.c:41]:
```suggestion
```

File: keyboards/boardsource/3x4/readme.md [hunk 9]
```diff
@@ -0,0 +1,14 @@
+# 3x4
+![Boardsource 3x4 macro](https://i.imgur.com/EiPF1hh.jpg)
+
+this macro pad / small 12 key was inspired by the plaid look
+
+Keyboard Maintainer: [Boardsource](https://github.com/daysgobye)  
+Hardware Supported: 3x4 v1
+Hardware Availability: this is macro pad is available from the Boardsource store
```
Comment (tzarc) [inline keyboards/boardsource/3x4/readme.md:8]: Please add a link.

File: keyboards/boardsource/3x4/rules.mk [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (tzarc) [inline keyboards/boardsource/3x4/rules.mk:11]:
```suggestion
```

**Comment by tzarc** [comment]
Just confirming, it's `boardsource`, not `bordsource`, right?

File: keyboards/boardsource/3x4/readme.md (hunk 9 above)
Comment (daysgobye) [reply keyboards/boardsource/3x4/readme.md]: Ok I did it. Thank you very much and good catch.

File: keyboards/boardsource/3x4/rules.mk [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality on B7 by default
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
```
Comment (fauxpark) [inline keyboards/boardsource/3x4/rules.mk:32]:
```suggestion
BOOTMAGIC_ENABLE = lite      # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = yes        # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
MIDI_ENABLE = no            # MIDI support
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```

File: keyboards/boardsource/3x4/rules.mk [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+LAYOUTS = 3x4
```
Comment (fauxpark) [inline keyboards/boardsource/3x4/rules.mk:34]:
```suggestion
```

File: keyboards/boardsource/3x4/readme.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
+# 3x4
+![Boardsource 3x4 macro](https://i.imgur.com/EiPF1hh.jpg)
+
+this macro pad / small 12 key was inspired by the plaid look
+
+Keyboard Maintainer: [Boardsource](https://github.com/daysgobye)  
+Hardware Supported: 3x4 v1
+Hardware Availability: this is macro pad is available from the [Boardsource store](https://boardsource.xyz/store/5ecc2008eee64242946c98c1)
```
Comment (fauxpark) [inline keyboards/boardsource/3x4/readme.md:8]:
```suggestion
* Keyboard Maintainer: [Boardsource](https://github.com/daysgobye)
* Hardware Supported: 3x4 v1
* Hardware Availability: [Boardsource store](https://boardsource.xyz/store/5ecc2008eee64242946c98c1)
```

File: keyboards/boardsource/3x4/config.h [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xF7E0
+#define PRODUCT_ID      0x0412
+#define DEVICE_VER      0x0000
+#define MANUFACTURER    Boardsource
+#define PRODUCT         3x4
+#define DESCRIPTION     Little macro pad
+
+/* key matrix size */
+#define MATRIX_ROWS 3
```
Comment (fauxpark) [inline keyboards/boardsource/3x4/config.h:16]:
```suggestion
#define MATRIX_ROWS 3
```

File: keyboards/boardsource/3x4/3x4.h [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+    K00, K01, K02, K03,  \
+    K10, K11, K12, K13,  \
+    K20, K21, K22, K23 \
+    ) { \
+        {K00, K01, K02, K03}, \
+        {K10, K11, K12, K13}, \
+        {K20, K21, K22, K23} \
+    }
+
+// #endif
```
Comment (fauxpark) [inline keyboards/boardsource/3x4/3x4.h:14]:
```suggestion
```

**Pull request merged**